### PR TITLE
chore(travis): add e2e tests

### PR DESCRIPTION
**Pull request by actionless**

### CODE REVIEW

**Comment by actionless** [comment]
@petrjasek @amagdas 
can u take a look on build logs?
i tried with both chrome and firefox but anyway from 2 to 5 e2e tests are failing from time to time

screenshots of failed tests:
http://curl.io/get/cyusjvxu/0dbc509e10c16dc820f5609137c39cd72d7f4e5c

File: client/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "karma-ng-html2js-preprocessor": "^0.1.0",
     "lodash": "^2.4.1",
     "karma-coverage": "^0.2.0",
-    "protractor": "~1.6.1",
+    "protractor": "actionless/protractor#old-chrom",
```
Comment (amagdas) [inline client/package.json]: With latest protractor are we having more errors?

Comment (actionless) [inline client/package.json]: it's the latest protractor but the thing what chromedriver version is hardcoded and in order to use travis we need older version of chromedriver (travis have chromium 37 installed while newer chromedriver requires 38)

and i docker-based travis workers we can't do operations which require super user rights so it's not possible to install newer chrome

**Comment by actionless** [comment]
@plamut come and join us

File: client/spec/setup.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ beforeEach(function(done) {
     getToken(function() {
         resetApp(function() {
             browser.driver.getCurrentUrl().then(function(url) {
-                if (url.indexOf('data:') !== 0) {
+                if (url.indexOf('http') === 0) {
```
Comment (actionless) [inline client/spec/setup.js]: that's for compatibility with firefox

**Comment by plamut** [comment]
Can this be reproduced locally? Or does the issue only appear on Travis?

**Comment by actionless** [comment]
only on travis, but now i am trying to fully reproduce the environment (chrome and chromedriver versions at least) to check if the problem can be related to any of them

**Comment by actionless** [comment]
@plamut from first checks i have an impression what it can be chromedriver version, but an exception is different than here:

```
17-Feb-2015 10:35:11 	Uncaught exception: Timed out waiting for Protractor to synchronize with the page after 11 seconds. Please see https://github.com/angular/protractor/blob/master/docs/faq.md
```

if it would be clear that chromedriver is the root cause when it seems like we will have no other choice than stick back to openVZ-based builds (they're slower a bit)

**Comment by actionless** [comment]
i will merge Petr's changes and try again

**Comment by plamut** [comment]
OK, thanks for the update(s).

EDIT: it seems tha the build has just passed?

**Comment by petrjasek** [comment]
I think it would be good to put this shell code from `.travis.yml` into `.sh` files so we could run it locally

**Comment by actionless** [comment]
oops, just noticed what because of some networking glitch my comment was not being sent:

sequence for running e2e tests relies on travis build lifetime hooks (running xvfb in before_install and starting servers in before_script) -- moving it outside as separate scripts will not be very helpful so i better recommend using this tool for local reproduce of travis build: 
https://github.com/grosser/wwtd

**Comment by actionless** [comment]
@petrjasek also as a separate script it's presented as part of docker pr:
https://github.com/actionless/superdesk/blob/add-docker-ci-scripts/docker/run_e2e_tests.sh

**Comment by petrjasek** [comment]
ok makes sense

**Comment by petrjasek** [comment]
:+1:

**Comment by actionless** [comment]
@amagdas please review as well, it need to be merged asap

**Comment by amagdas** [comment]
:+1:

**Pull request merged**